solr/solrj/src/test/org/apache/solr/client/solrj/impl/LB2SolrClientTest.java

```diff
@@ -196,9 +197,18 @@ public void testTwoServers() throws Exception {
       solr[1].jetty = null;
       startJettyAndWaitForAliveCheckQuery(solr[0]);
 
-      resp = h.lbClient.query(solrQuery);
-      name = resp.getResults().get(0).getFieldValue("name").toString();
-      assertEquals("solr/collection10", name);
+      RetryUtil.retryOnBoolean(
+          5000,
+          500,
+          () -> {
+            try {
+              QueryResponse resp1 = h.lbClient.query(solrQuery);
+              String name1 = resp1.getResults().get(0).getFieldValue("name").toString();
+              return ("solr/collection10".equals(name1));
```

Comment:
instead of adding 1 to the end... really it's fine do do this whole thing as one expression (inline each var).